src/components/Modal/Modal.tsx

```diff
@@ -64,178 +64,156 @@ export interface ModalProps extends FooterProps {
 }
 type CombinedProps = ModalProps & WithAppProviderProps;
 
-interface State {
-  iframeHeight: number;
-}
-
 const getUniqueID = createUniqueIDFactory('modal-header');
 
-class Modal extends React.Component<CombinedProps, State> {
-  static Section = Section;
-
-  state: State = {
-    iframeHeight: IFRAME_LOADING_HEIGHT,
-  };
-
-  private headerId = getUniqueID();
-  private activatorRef = React.createRef<HTMLDivElement>();
-
-  render() {
-    const {
-      children,
-      title,
-      src,
-      iFrameName,
-      open,
-      instant,
-      sectioned,
-      loading,
-      large,
-      limitHeight,
-      footer,
-      primaryAction,
-      secondaryActions,
-      polaris: {intl},
-      onScrolledToBottom,
-      activator,
-      onClose,
-    } = this.props;
-
-    const {iframeHeight} = this.state;
-
-    const iframeTitle = intl.translate('Polaris.Modal.iFrameTitle');
-
-    let dialog: React.ReactNode;
-    let backdrop: React.ReactNode;
-    if (open) {
-      const footerMarkup =
-        !footer && !primaryAction && !secondaryActions ? null : (
-          <Footer
-            primaryAction={primaryAction}
-            secondaryActions={secondaryActions}
-          >
-            {footer}
-          </Footer>
-        );
-
-      const content = sectioned
-        ? wrapWithComponent(children, Section, {})
-        : children;
-
-      const body = loading ? (
-        <div className={styles.Spinner}>
-          <Spinner />
-        </div>
-      ) : (
-        content
-      );
-
-      const bodyMarkup = src ? (
-        <iframe
-          name={iFrameName}
-          title={iframeTitle}
-          src={src}
-          className={styles.IFrame}
-          onLoad={this.handleIFrameLoad}
-          style={{height: `${iframeHeight}px`}}
-        />
-      ) : (
-        <Scrollable
-          shadow
-          className={styles.Body}
-          onScrolledToBottom={onScrolledToBottom}
+const Modal: React.FunctionComponent<CombinedProps> & {
+  Section: typeof Section;
+} = function Modal({
+  children,
+  title,
+  src,
+  iFrameName,
+  open,
+  instant,
+  sectioned,
+  loading,
+  large,
+  limitHeight,
+  footer,
+  primaryAction,
+  secondaryActions,
+  polaris: {intl},
```

Comment:
A big part of moving towards functional components is that we can remove the withAppProvider HoC that provides this polaris prop. We've got a handful of custom hooks that replace these values of polaris

Remove this and the withAppProvider Hoc and instead do:

```
const i18n = useI18n();

// ...

i18n.translate('Polaris.SomeKey');
```